ui/src/app/components/workflows/workflow-form/jobs/job/job-types/hyperdrive-job/hyperdrive-job.component.ts

```diff
@@ -27,20 +37,76 @@ export class HyperdriveJobComponent implements OnInit, OnDestroy {
   @Input() jobParameters: HyperdriveDefinitionParametersModel;
   @Output() jobParametersChange: EventEmitter<HyperdriveDefinitionParametersModel> = new EventEmitter();
   @Input() isJobTemplateSelected: boolean;
-  @Input() jobTemplateChanges: EventEmitter<string>;
+  @Input() jobTemplateChanges: EventEmitter<JobTemplateChangeEventModel>;
+  @Input() jobTemplateParameters: HyperdriveTemplateParametersModel;
+
+  private hyperdriveType: string = undefined;
 
   jobTemplateChangesSubscription: Subscription;
 
+  hyperdriveTypes = hyperdriveTypes;
+  hyperdriveTypesMap = hyperdriveTypesMap;
+  hyperdriveFields = hyperdriveFields;
+
   constructor() {
     // do nothing
   }
 
   ngOnInit(): void {
-    this.jobTemplateChangesSubscription = this.jobTemplateChanges.subscribe((value) => {
-      if (value) {
-        this.jobParametersChange.emit({ ...this.jobParameters, jobJar: undefined, mainClass: undefined });
+    this.jobTemplateChangesSubscription = this.jobTemplateChanges.subscribe((value: JobTemplateChangeEventModel) => {
+      this.jobParametersChange.emit({
+        ...this.jobParameters,
+        jobJar: undefined,
+        mainClass: undefined,
+        appArguments: this.getAppArguments(),
```

Comment:
If you change the job template, then the source topic is lost. I think it's because you filter the job template arguments out here.

Reply:
I made it simpler, I just clean all app arguments on hyperdrive type or job template change.
If we decide to keep app arguments we can make it later

Reply:
I should not have said anything, this was one of my favorite features 😅

Reply:
Oh :/ It's back